### DISCUSSION TITLE
Is it possible for Quarto's search to exclude certain parts of a web page from the search index?

### DISCUSSION THREAD

**Discussion by nickvigilante** [quarto-dev/quarto-cli#11186] [Q&A] [answered]
### Description

We have several pages across our docs that embed Jupyter notebook cells, and the Quarto search functionality indexes the output of these cells by default, as I would expect it to. However, there are a lot of unnecessary search results when I try to search for a term contained in one of these cell outputs.

Is there an OOTB way via the default Quarto search to suppress a specific div or span from the search results, either via class or shortcode option?

Some ideas of what I'd be looking for are as follows:

```
::: {.search-ignore}

{{< embed path/to/notebook.ipynb#cell echo=true >}}

:::
```

```
<!-- search off -->

{{< embed path/to/notebook.ipynb#cell echo=true >}}

<!-- search on -->
```

```
{{< embed path/to/notebook.ipynb#cell echo=true search-index=false >}}
```

I know Algolia can exclude certain parts of a web page from the search index, but I thought I'd see if the default search can do the same.

**Comment by mcanouil** [comment]
See <https://quarto.org/docs/websites/website-search.html>, probably the "tagFilters".

**Reply by nickvigilante** [reply]
It seems like it doesn't like the addition of `params` without using it in Algolia:

```
ERROR: Project _quarto.yml validation failed.

In file _quarto.yml
(line 28, columns 5--11) property name params is invalid
27:     show-item-context: tree
28:     params:
       ~~~~~~~
29:       tagFilters:
ℹ The error happened in location website:search:params.


Stack trace:

In file _quarto.yml
(line 28, columns 5--11) property name params is invalid
27:     show-item-context: tree
28:     params:
       ~~~~~~~
29:       tagFilters:
ℹ The error happened in location website:search:params.

    at readAndValidateYamlFromFile (file:///Applications/quarto/bin/quarto.js:20497:15)
    at async file:///Applications/quarto/bin/quarto.js:73752:26
    at async projectContext (file:///Applications/quarto/bin/quarto.js:73546:26)
    at async Command.actionHandler (file:///Applications/quarto/bin/quarto.js:100720:25)
    at async Command.execute (file:///Applications/quarto/bin/quarto.js:8017:13)
    at async Command.parseCommand (file:///Applications/quarto/bin/quarto.js:7907:20)
    at async quarto (file:///Applications/quarto/bin/quarto.js:118224:9)
    at async file:///Applications/quarto/bin/quarto.js:118244:9
    at async mainRunner (file:///Applications/quarto/bin/quarto.js:118128:9)
    at async file:///Applications/quarto/bin/quarto.js:118235:5```

**Reply by mcanouil** [reply]
If you don't share the original code, the error message does not help much unfortunately.

**Reply by nickvigilante** [reply]
Sorry, here you go:

```
website:
  navbar:
    search: true
    background: light
    collapse-below: lg
    pinned: false
  search:
    show-item-context: tree
    params:
      tagFilters:
        - 'exclude-from-search'
```

**Reply by cscheid** [reply]
Ah, it's a documentation bug. That field needs to be inside `algolia`.

**Reply by cscheid** [reply]
I just fixed our webpage, sorry about that: https://github.com/quarto-dev/quarto-web/commit/269342b135b52cbb7c854ad1c86cdac44bbd087e

Thanks for letting us know about this.

**Reply by nickvigilante** [reply]
Yep, sure thing. But do you have any other suggestions about how I might be able to work around our problem? I thought about the comment approach (`<!-- search on -->`/`<!-- search off -->`), but I realize we strip comments, so that might not work. Wondering if this might just be an enhancement we log to add a class that excludes a specific `div` or `span` from being searched. What are your thoughts?

**Reply by cscheid** [reply]
I don't have a suggestion, no. You can file an enhancement request.

**Reply by mcanouil** [reply]
Note that you can already add classes to code cells or eventually surround a code cell with a div.
Not sure how to tell Algolia to exclude those though as it seems `tagFilters` is to filter in not out.

**Reply by nickvigilante** [reply]
We're not using Algolia, we're using the OOTB Quarto search. I'm trying to figure out how to exclude certain blocks from the OOTB Quarto search.

**Reply by cscheid** [reply]
FWIW the code that builds/updates Quarto's search indexes is [src/project/types/website/website-search.ts](https://github.com/quarto-dev/quarto-cli/tree/main/src/project/types/website/website-search.ts), specifically `updateSearchIndex`.  That would be the place for a PR to start.

**Comment by mcanouil** [comment, marked as answer]
Tracked in:

- #11189
